plugins/Microsoft.DocAsCode.Build.UniversalReference/UniversalReferenceDocumentProcessor.cs

```diff
@@ -0,0 +1,250 @@
+                                into g
+                                select g.First()).ToImmutableArray();
+            result.ExternalXRefSpecs = GetXRefFromReference(vm).ToImmutableArray();
+            UpdateModelContent(model);
+
+            return result;
+        }
+
+        #endregion
+
+        #region Protected/Private Methods
+
+        protected virtual void UpdateModelContent(FileModel model)
+        {
+            // to be implemented
+        }
+
+        private IEnumerable<XRefSpec> GetXRefFromReference(PageViewModel vm)
+        {
+            if (vm.References == null)
+            {
+                yield break;
+            }
+            foreach (var reference in vm.References)
+            {
+                if (reference != null && reference.IsExternal != false)
+                {
+                    yield return GetXRefSpecFromReference(reference);
+                }
+            }
+        }
+
+        private static IEnumerable<XRefSpec> GetXRefInfo(
+            ItemViewModel item,
+            string key,
+            List<ReferenceViewModel> references)
+        {
+            var result = new XRefSpec
+            {
+                Uid = item.Uid,
+                Name = item.Name,
+                Href = ((RelativePath)key).UrlEncode().ToString(),
+                CommentId = item.CommentId,
+            };
+            if (item.Names.Count > 0)
+            {
+                foreach (var pair in item.Names)
+                {
+                    result["name." + pair.Key] = pair.Value;
+                }
+            }
+            if (!string.IsNullOrEmpty(item.FullName))
+            {
+                result["fullName"] = item.FullName;
+            }
+            if (item.FullNames.Count > 0)
+            {
+                foreach (var pair in item.FullNames)
+                {
+                    result["fullName." + pair.Key] = pair.Value;
+                }
+            }
+            if (!string.IsNullOrEmpty(item.NameWithType))
+            {
+                result["nameWithType"] = item.NameWithType;
+            }
+            if (item.NamesWithType.Count > 0)
+            {
+                foreach (var pair in item.NamesWithType)
+                {
+                    result["nameWithType." + pair.Key] = pair.Value;
+                }
+            }
+            yield return result;
+
+            // generate overload xref spec.
+            if (item.Overload != null)
+            {
+                var reference = references.Find(r => r.Uid == item.Overload);
+                if (reference != null)
+                {
+                    yield return GetXRefInfo(reference, key);
+                }
+            }
+        }
+
+        private static XRefSpec GetXRefInfo(ReferenceViewModel item, string key)
+        {
+            var result = GetXRefSpecFromReference(item);
+            result.Href = ((RelativePath)key).UrlEncode().ToString();
```

Comment:
item.Href is already encoded

---
In reply to: [112601786](https://github.com/dotnet/docfx/pull/1574#discussion_r112601786) [](ancestors = 112601786)